sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/StreamExecution.scala

```diff
@@ -317,15 +358,18 @@ class StreamExecution(
     // TODO: Move this to IncrementalExecution.
 
     // Request unprocessed data from all sources.
-    val newData = availableOffsets.flatMap {
-      case (source, available)
+    val newData = timeIt(GET_BATCH_LATENCY) {
```

Comment:
Yeah. The intention in GET_BATCH_LATENCY is to measure the time taken in the non-lazy part.